drivers/char/tpm/tpm_tis_spi_main.c

```diff
@@ -231,6 +234,38 @@ static int tpm_tis_spi_probe(struct spi_device *dev,
 	return tpm_tis_spi_init(dev, phy, irq, phy_ops);
 }
 
+static int tpm_tis_slb9670_unset_reset(struct tpm_tis_data *data)
+{
+	/* perform reset sequence */
+	gpiod_set_value(data->reset_gpio, 1);
+	msleep(SLB9670_TIME_RSTIN);
+	gpiod_set_value(data->reset_gpio, 0);
+	udelay(SLB9670_TIME_WRST);
+	gpiod_set_value(data->reset_gpio, 1);
+	msleep(SLB9670_TIME_RSTIN);
+	gpiod_set_value(data->reset_gpio, 0);
+	udelay(SLB9670_TIME_WRST);
+	gpiod_set_value(data->reset_gpio, 1);
+	msleep(SLB9670_TIME_RSTIN);
+
+	return 0;
+}
+
+static int tpm_tis_slb9670_set_reset(struct tpm_tis_data *data)
+{
+	gpiod_set_value(data->reset_gpio, 0);
```

Comment:
If we would have a sequence of set_reset(); unset_reset() we might not wait the 2µs the reset must be asserted before we deassert it again. But I guess this is more of a theoretical issue then a reallife problem.

Reply:
Such a sequence is not possible (at least not within 2 usecs) since set_reset() is only called at system shutdown.